internal/server/grants.go

```diff
@@ -145,33 +145,55 @@ func (a *API) UpdateGrants(c *gin.Context, r *api.UpdateGrantsRequest) (*api.Emp
 	iden := access.GetRequestContext(c).Authenticated.User
 	var addGrants []*models.Grant
 	for _, g := range r.GrantsToAdd {
-		grant := getGrantFromGrantRequest(g)
+		grant, err := getGrantFromGrantRequest(c, g)
+		if err != nil {
+			return nil, err
+		}
 		grant.CreatedBy = iden.ID
 		addGrants = append(addGrants, grant)
 	}
 
 	var rmGrants []*models.Grant
 	for _, g := range r.GrantsToRemove {
-		grant := getGrantFromGrantRequest(g)
+		grant, err := getGrantFromGrantRequest(c, g)
+		if err != nil {
+			return nil, err
+		}
 		rmGrants = append(rmGrants, grant)
 	}
 
 	return nil, access.UpdateGrants(c, addGrants, rmGrants)
 }
 
-func getGrantFromGrantRequest(r api.GrantRequest) *models.Grant {
+func getGrantFromGrantRequest(c *gin.Context, r api.GrantRequest) (*models.Grant, error) {
 	var subject uid.PolymorphicID
 
 	switch {
+	case r.Name != "":
+		// lookup user name
+		identity, err := access.GetIdentityByName(c, r.Name)
+		if err != nil {
+			return nil, fmt.Errorf("%w: couldn't find user %s", internal.ErrBadRequest, r.Name)
```

Comment:
This could also be a result of insufficient permissions I believe

Reply:
I switched this to look for if the userName (and now also groupName) was not found. It will still pass back the insufficient permissions which isn't tbh the best error given it might not be clear *why* it's trying to look up an identity (or group).

Ideally we would automatically give everyone the View permission though by default which would solve this most of the time.